code/game/objects/objs.dm

```diff
@@ -48,6 +48,11 @@ var/global/list/reagents_to_log = list(FUEL, PLASMA, PACID, SACID, AMUTATIONTOXI
 	if(src in processing_objects)
 		processing_objects -= src
 
+	if(integratedpai)
+		integratedpai.forceMove(get_turf(src))
```

Comment:
Not necessarily.

In that case I'm more in favor of handling the case of the MULE being shot as such that the pAI gets ejected.